src/field_theory/minpoly/is_integrally_closed.lean

```diff
@@ -140,6 +140,17 @@ begin
   { rw [(monic hs).leading_coeff, hmo.leading_coeff] }
 end
 
+theorem prime_of_is_integrally_closed {x : S} (hx : is_integral R x) :
```

Comment:
You can maybe add an instance `is_integrally_closed` to any field (if I am not missing some stupid stuff this is trivially true) and then this can replace [minpoly.prime](https://leanprover-community.github.io/mathlib_docs/field_theory/minpoly/field.html#minpoly.prime).

Reply:
I think we should add the instance from `unique_factorization_monoid` to `is_integrally_closed`, by combining  and [gcd_monoid.to_is_integrally_closed](https://leanprover-community.github.io/mathlib_docs/algebra/gcd_monoid/integrally_closed.html#gcd_monoid.to_is_integrally_closed) and [unique_factorization_monoid.to_gcd_monoid](https://leanprover-community.github.io/mathlib_docs/ring_theory/unique_factorization_domain.html#unique_factorization_monoid.to_gcd_monoid) (the latter can't be an instance because `gcd_monoid` is data), since there's already [division_ring.is_principal_ideal_ring](https://leanprover-community.github.io/mathlib_docs/ring_theory/principal_ideal_domain.html#division_ring.is_principal_ideal_ring) and [principal_ideal_ring.to_unique_factorization_monoid](https://leanprover-community.github.io/mathlib_docs/ring_theory/principal_ideal_domain.html#principal_ideal_ring.to_unique_factorization_monoid).

Reply:
Sure I'll do that

Reply:
If we can replace `minpoly.prime` with this new version (I wouldn't change the name) even better. Just be sure it works for fields.

Reply:
Yes, that's always fine. There is a building error though.

Reply:
I'll fix that as soon as I'm back on my laptop;)